src/Cryptol/Eval/Value.hs

```diff
@@ -122,15 +132,22 @@ data GenValue sym
   | VInteger !(SInteger sym)                   -- ^ @ Integer @ or @ Z n @
   | VRational !(SRational sym)                 -- ^ @ Rational @
   | VFloat !(SFloat sym)
-  | VSeq !Integer !(SeqMap sym (GenValue sym)) -- ^ @ [n]a   @
-                                               --   Invariant: VSeq is never a sequence of bits
+  | VSeqCtor !Integer !(SeqMap sym (GenValue sym)) -- ^ @ [n]a   @
+                                                   --   Invariant: VSeq is never a sequence of bits
   | VWord !(WordValue sym)                     -- ^ @ [n]Bit @
   | VStream !(SeqMap sym (GenValue sym))       -- ^ @ [inf]a @
   | VFun  CallStack (SEval sym (GenValue sym) -> SEval sym (GenValue sym)) -- ^ functions
   | VPoly CallStack (TValue -> SEval sym (GenValue sym))   -- ^ polymorphic values (kind *)
   | VNumPoly CallStack (Nat' -> SEval sym (GenValue sym))  -- ^ polymorphic values (kind #)
  deriving Generic
 
+pattern VSeq :: Integer -> SeqMap sym (GenValue sym) -> GenValue sym
+pattern VSeq len vals <- (VSeqCtor len vals)
+
+{-# COMPLETE VRecord, VTuple, VEnum, VBit, VInteger,
+             VRational, VFloat, VWord, VStream,
+             VFun, VPoly, VNumPoly, VSeq #-}
```

Comment:
Worth leaving a comment here that this should include all of `GenValue`'s data constructors _except_ for `VSeqCtor`, which has instead been swapped out for `VSeq`. That way, if we ever need to add another `GenValue` data constructor later, it will be more obvious that we need to update this `COMPLETE` set (and how to do so).